src/server/common.h

```diff
@@ -117,6 +118,33 @@ struct OpArgs {
   }
 };
 
+// A strong type for a lock tag. Helps to disambiguide between keys and the parts of the
+// keys that are used for locking.
+class LockTag {
+  std::string_view str_;
+
+ public:
+  using is_stackonly = void;  // marks that this object does not use heap.
+
+  LockTag() = default;
+  explicit LockTag(std::string_view key);
+
+  explicit operator std::string_view() const {
```

Comment:
Nit: if you renamed this to `tag()`, you could add, in the future, a `key()` which returns the original (full) key. That way this class may be even more useful.

Reply:
I do not store the original key. The only purpose of this class is to be a distinct type to distinguish between key and the tags.